src/Attachment.tsx

```diff
@@ -25,6 +71,10 @@ const Media = (props: {
     const { type, ... mediaProps } = props; // this allows us to keep 'type' out of the final HTML
     switch (type) {
         case 'video':
+            if (isYoutube(mediaProps.src)) {
+                return <Youtube { ... mediaProps } />
+            }
+
```

Comment:
I tend to agree, since it may become a verbose block:
```
 if (isYouTube...)
   return <YouTube...>
 if (isVimeo...)
   return <Vimeo...>
 if (isFacebook...)
   return <Facebook...>
 if (isInstagram...)
   return <Instagram...>
```